### PR TITLE
feat(api): Upgrade Folder, License & Obligation APIs to Version 2

### PR DESCRIPTION
**Pull request by dvjsharma**
## Description

This PR aims to upgrade the `/folder` and `/license` endpoints to v2 in accordance with the new upgrade guidelines.

### Changes

* Completed all tasks mentioned in #2710.
* Updated `openapiv2.yaml` to have all documentation changes.
* Modified models to make them version aware.

### How to Test

* Open the `openapiv2.yaml` file in Swagger Editor and test all the updated endpoints for v2 specific responses. All function as intended.
* Execute unit tests, all pass without any issues.

CC @GMishx @shaheemazmalmmd @dushimsam

### CODE REVIEW

**Comment by dvjsharma** [comment]
Few points to keep in mind regarding some upgrades: 

- I focused solely on resolving failing controller tests without making them version aware since many of them do not specifically assess the semantics of request or response objects, but rather the functionality, which remains largely consistent between v1 and v2. We can expand this coverage to accommodate v2 tests as well. But, creating separate tests with only a difference in parameter acceptance would result in a significant amount of duplicated code. Need thoughts on this. (refer [this](https://github.com/fossology/fossology/blob/master/src/www/ui_tests/api/Models/UploadTest.php), makes sense for models but not controllers. In-line integration might be helpful).

- Changing `file_input` to camelCase might break a lot of code. Will update more on this. (a thing left in this PR, will do)

- Migration of `groupName` from the headers to query parameters has already been globally completed (via Middleware) in #2633, so didn't touch it.

- for `/license/adminacknowledgements` GET & `/license/stdcomments` GET, response models do not exist. Although I did the change but need advice on should I create new models for them. (these are specific endpoints and new model won't be used anywhere, but to keep things consistent, IMO new model should be made).

Apart from these points, everything functions as anticipated.

**Comment by GMishx** [comment]
Sorry for such a late response.

> Few points to keep in mind regarding some upgrades:
> 
>     * for `/license/adminacknowledgements` GET & `/license/stdcomments` GET, response models do not exist. Although I did the change but need advice on should I create new models for them. (these are specific endpoints and new model won't be used anywhere, but to keep things consistent, IMO new model should be made).

Having models for them as well, it is good keeping things consistent.

>     * Should I also modify `/folder` endpoints to use unique names instead folder IDs?

Folder names are not unique, thus they can operate on IDs. Also, the endpoints expose he folder ids in the response fields.

File: src/www/ui/api/Controllers/LicenseController.php [hunk 1]
```diff
@@ -473,7 +483,8 @@ public function getAllAdminAcknowledgements($request, $response, $args)
     foreach ($res as $key => $ack) {
       $res[$key]['id'] = intval($ack['la_pk']);
       unset($res[$key]['la_pk']);
-      $res[$key]['is_enabled'] = $ack['is_enabled'] == "t";
+      $res[$key]['isEnabled'] = $ack['is_enabled'] == "t";
+      unset($res[$key]['is_enabled']);
```
Comment (GMishx) [inline src/www/ui/api/Controllers/LicenseController.php:487]: The field `is_enabled` is still required for v1 response, right?

File: src/www/ui/api/Controllers/LicenseController.php [hunk 2]
```diff
@@ -578,8 +589,9 @@ public function getAllLicenseStandardComments($request, $response, $args)
     $res = $this->licenseStdCommentDao->getAllComments();
     foreach ($res as $key => $ack) {
       $res[$key]['id'] = intval($ack['lsc_pk']);
-      $res[$key]['is_enabled'] = $ack['is_enabled'] == "t";
+      $res[$key]['isEnabled'] = $ack['is_enabled'] == "t";
       unset($res[$key]['lsc_pk']);
+      unset($res[$key]['is_enabled']);
```
Comment (GMishx) [inline src/www/ui/api/Controllers/LicenseController.php:594]: The field `is_enabled` is still required for v1 response, right?

**Review by GMishx — CHANGES_REQUESTED** [review]
Minor changes needed.

File: src/www/ui/api/Controllers/LicenseController.php (hunk 1 above)
Comment (dvjsharma) [reply src/www/ui/api/Controllers/LicenseController.php:487]: Yes, I will create a new model for this response and address it there.

**Comment by dvjsharma** [comment]
Hey @GMishx, I have added the following based on the last review:
1. The `/license/adminacknowledgements` GET and `/license/stdcomments` GET endpoints now each have their own models.
2. The `is_enabled` version bug is fixed.
3. Completed additional tasks recently added to #2710 

CC @dushimsam

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Review by GMishx — APPROVED** [review]
Tested, working as expected.

**Pull request merged**